napari/layers/base/base.py

```diff
@@ -1545,12 +1545,16 @@ def _update_draw(
 
         else:
             # The stored corner_pixels attribute must contain valid indices.
-            displayed_extent = self.extent.data[:, displayed_axes]
-            data_bbox_clipped = np.clip(
-                data_bbox_int, displayed_extent[0], displayed_extent[1]
-            )
             corners = np.zeros((2, self.ndim), dtype=int)
-            corners[:, displayed_axes] = data_bbox_clipped
+            # Some empty layers (e.g. Points) may have a data extent that only
+            # contains nans, in which case the integer valued corner pixels
+            # cannot be meaningfully set.
+            displayed_extent = self.extent.data[:, displayed_axes]
+            if not np.all(np.isnan(displayed_extent)):
```

Comment:
Ideally, I'd want to have something like `if self.is_empty()` here instead. The existing code effectively serves as a proxy for that an is deliberately distinct from checking if there any nans in `displayed_extent` (which will also cause the same warning to be issued).

Reply:
```suggestion
            if not np.any(np.isnan(displayed_extent)):
```
I think that the existence `nan` on nay position should prevent this.

Reply:
I did consider this as a more general fix (i.e. to prevent the warning), but my concern is that it might hide other bugs.

Like I mentioned above, I really want something like `Layer.is_empty()` in which case layer extent and corner pixels are not well defined. If we have one nan in the layer extent (i.e. instead of all nans), something probably went very wrong. Though in that case, I'd probably expect worse things to happen, so I'd also be fine with this suggestion.